metis-indexing/src/main/java/eu/europeana/indexing/utils/SetUtils.java

```diff
@@ -19,12 +17,48 @@ private SetUtils() {}
 
   /**
    * <p>
-   * This method generates all possible combinations of elements of different sets. Each combination
-   * consists of zero or one element of each set. As such, the number of combinations of sets s1,
-   * s2, ..., sn that will be generated is equal to (|s1| + 1) * (|s2| + 1) * ... * (|sn| + 1),
-   * where |s| denotes the cardinality (size) of set s. However, depending on the choice of the
-   * types and the concatenator, not all of these may be unique and the result set may contain fewer
-   * items.
+   * This method generates all possible combinations of elements of different sets.
+   * </p>
+   * <p>
+   * Each combination consists of zero or one element of each set. As such, the number of
+   * combinations of sets s1, s2, ..., sn that will be generated is equal to (|s1| + 1) * (|s2| + 1)
+   * * ... * (|sn| + 1), where |s| denotes the cardinality (size) of set s. However, depending on
+   * the choice of the types and the concatenator, not all of these may be unique and the result set
+   * may contain fewer items.
+   * </p>
+   * <p>
+   * It follows that if one of the input sets is empty, the result will be as if that set was not
+   * included in the options at all.
+   * </p>
+   *
+   * @param input The input sets. They will be processed in the order given here. Input sets can be
+   * empty, but not null.
+   * @param emptyCombination The empty combination (used as a seed for the algorithm).
+   * @param concatenator The function that appends/concatenates an element to a combination. This
+   * function <b>must</b> generate a new instance of the combination (rather than return the same
+   * instance).
+   * @param <E> The type of the individual elements (input).
+   * @param <C> The type of the combinations (output).
```

Comment:
Is this an "emtpyCombination"? It's more like an item that shouldn't be null to start the algorithm properly, right?
Same goes for the other methods

Reply:
Well, it is the  as an empty combination: it is a